source/blog/2017-03-14-a-tale-of-tempest-rpm-with-installers.html.md

```diff
@@ -0,0 +1,70 @@
+to maintain patches backports starting from ocata release.
+
+With this change, we have moved config_tempest.py script from forked tempest to separate project [python-tempestconf](https://github.com/redhat-openstack/python-tempestconf)
+so that anyone can use it with any OpenStack installers to generate tempest config automatically and packaged in [RDO](https://review.rdoproject.org/r/#/q/topic:add-tempestconf) also.
+
+What have we done to make a happy integration between tempest rpm and the installers?
+
+Currently, puppet-openstack-integration, packstack, and tripleo-quickstart heavily use RDO packages. So using tempest rpm with these installers will be the best match.
+Before starting the integration, we need to make the initial ground ready. Till newton release, all these installers are using Tempest from Source in their respective CI.
+We have started the match making of tempest rpm with installers.
+puppet-openstack-integration and packstack consume puppet-modules. So in order to consume tempest rpm, first I need to fix the [puppet-tempest](https://github.com/openstack/puppet-tempest).
+
+## [puppet-tempest](https://github.com/openstack/puppet-tempest)
+
+It is a puppet-module to install and configure tempest and openstack-services tempest plugins based on the services available from source as well as packages.
+So we have fixed puppet-tempest to install tempest rpm from the package and created a tempest workspace. In order to use that feature through puppet-tempest module [https://review.openstack.org/#/c/425085/].
+you need to add *install_from_source => 'false'* and *tempest_workspace => 'path to tempest workspace'* to tempest.pp and it will do the job for you.
+Now we are using the same feature in puppet-openstack-integration and packstack.
+
+## [puppet-openstack-integration](https://github.com/openstack/puppet-openstack-integration)
+
+It is a collection of scripts and manifests for puppet module testing (which powers the openstack-puppet CI).
+From Ocata release, we have added a flag TEMPEST_FROM_SOURCE flag in [run_tests.sh script](https://github.com/openstack/puppet-openstack-integration/blob/master/run_tests.sh).
+Just change TEMPEST_FROM_SOURCE to false in the [run_test.sh](https://review.openstack.org/#/c/427578/), Tempest is installed and configured from packages using puppet-tempest.
+
+## [packstack](https://github.com/openstack/packstack)
+
+It is a utility to install OpenStack on Red Hat based operating system. Till newton, Tempest is installed and ran by packstack from the upstream source and behind the scenes, puppet-tempest does the job for us. From Ocata, we have replaced this feature by using [tempest RDO package](https://review.openstack.org/#/c/428102/).You can use this feature by running following command: 
+
+	$ sudo packstack --allinone --config-provision-tempest=y.
```

Comment:
With the above command, we just provision tempest, not run it. You need to add --run-tempest=y to get that.